refinery/file_store/models.py

```diff
@@ -177,24 +171,11 @@ def save(self, *args, **kwargs):
             else:
                 self.filetype = extension.filetype
 
-        if self.datafile:
-            # symlink datafile if necessary
-            if (not self.is_local() and os.path.isabs(self.source) and
-                    settings.REFINERY_DATA_IMPORT_DIR not in self.source):
-                self._symlink_datafile()
-        else:
-            # delete file
-            try:
-                old_instance = FileStoreItem.objects.get(pk=self.pk)
-            except FileStoreItem.DoesNotExist:
-                pass  # this is a newly created instance
-            except FileStoreItem.MultipleObjectsReturned as exc:
-                logger.critical(
-                    "Error retrieving FileStoreItem with primary key '%s': %s",
-                    self.pk, exc
-                )
-            else:
-                old_instance.delete_datafile(save_instance=False)
+        # symlink datafile if possible
+        if (not self.datafile and os.path.isabs(self.source) and
```

Comment:
If this logic is required for a datafile to be symlinked, would it be better off residing within `_symlink_datafile`?

Reply:
No, this is a separate piece of functionality; `_symlink_datafile` only handles symlinking files on disk.

Reply:
Okay, but couldn't `_symlink_datafile` get used elsewhere and improperly symlink data files if the large `if` above it isn't copied over properly?

Reply:
That is why it is declared as a "private" function.

Reply:
Yes, but that doesn't prevent reuse within `FileStoreItem` (or anywhere else). I think that it makes sense that if there is a specific set of criteria for being able to symlink a data file, `_symlink_datafile` would handle it. Otherwise, this interface can potentially provide undesired functionality.

`_symlink_datafile` is already handling the case where the specified `source` isn't a file. Why wouldn't we move this logic within it as well?

Reply:
When a function is declared "private" it is not a part of the API, i.e. not to be used outside the file_store app.
What this does provide is an abstraction of low-level functionality of manipulating the file system from high-level functionality of saving a FileStoreItem instance. This makes the code easy to understand and test.

Reply:
Yes, I understand that...

Let's say someone calls `self._symlink_datafile()` within a new `FileStoreItem` method they introduce with the intent to symlink a datafile. They would need to remember to check:
```
if (not self.datafile and os.path.isabs(self.source) and settings.REFINERY_DATA_IMPORT_DIR 
    not in self.source and get_temp_dir() not in self.source)
```
beforehand, otherwise they could do something that wasn't originally intended, correct?